# **Phylogenomics and pangenome analysis**



*<u>Written by  Gamaliel López-Leal</u>*

*<u>Mail: gamaliel.lopez@docentes.uaem.edu.mx</u>*



### Get homologous groups and usnig SGF approach

In this protocol we start from genome sequence (.fna files). For the following practice find these files in the path: `/home/gama/Omicas/Genomes`. First we run `prokka` to annotate the selected genomes, for this we used this command line:

```
prokka infile.fna --kingdom Bacteria --outdir $file.dir --locustag $file
```

If you want to run prokka in a single process. I have written a script that calls all the fna files and runs prokka for each of the files.

```
nohup perl Running_PROKKA_toMultipleGenomes.pl &
```

Then you need to collect all .gff files

```
for subdir in *; do cp $subdir/*.gff /path/$subdir.gff; done;
```

### Orthologous Single Gene Families (using roary)

In order to get the SGF we used Roary with the following parameters.

```
nohup roary  -b "blastp -qcov_hsp_perc 90" -f Roary-Results_80Identity-90Coverage  -i 80   -s  -e -n -v ../*.gff -p 10 &
```

There are many interesting files in the roary output, for example. The accessory_graph.dot file generated by Roary is a Graphviz DOT representation of the accessory gene network of the analyzed pangenome.

```
dot -Tpng accessory_graph.dot -o accessory_graph.png
```

The same applies to the core_accessory_graph.dot file.

```
dot -Tpng core_accessory_graph.dot -o core_graph.png
```


If you want to use the gene presence-absence matrix to creat multiple files based on the HG information. First, you need to create a files using the information in the presence-absence matrix file and adding a unique tag in the first column. The file should be like this:


Additionally, we perform several plots in R according to the Roary manual (using the Roary out files):

```R
library(ggplot2)


mydata = read.table("number_of_new_genes.Rtab")
png(filename = "number_new_genes.png", width = 6, height = 4, units = "in", res = 300)
boxplot(mydata, data=mydata, main="Number of new genes",
         xlab="No. of genomes", ylab="No. of genes",varwidth=TRUE, ylim=c(0,max(mydata)), outline=FALSE)
dev.off()

mydata = read.table("number_of_conserved_genes.Rtab")
png(filename = "number_of_conserved_genes.png", width = 6, height = 4, units = "in", res = 300)
boxplot(mydata, data=mydata, main="Number of conserved genes",
          xlab="No. of genomes", ylab="No. of genes",varwidth=TRUE, ylim=c(0,max(mydata)), outline=FALSE)
dev.off()


######################

unique_genes = colMeans(read.table("number_of_unique_genes.Rtab"))
new_genes = colMeans(read.table("number_of_new_genes.Rtab"))

genes = data.frame( genes_to_genomes = c(unique_genes,new_genes),
                    genomes = c(c(1:length(unique_genes)),c(1:length(unique_genes))),
                    Key = c(rep("Unique genes",length(unique_genes)), rep("New genes",length(new_genes))) )
                    
ggplot(data = genes, aes(x = genomes, y = genes_to_genomes, group = Key, linetype=Key)) +geom_line()+
theme_classic() +
ylim(c(1,max(unique_genes)))+
xlim(c(1,length(unique_genes)))+
xlab("No. of genomes") +
ylab("No. of genes")+ theme_bw(base_size = 16) +  theme(legend.justification=c(1,1),legend.position=c(1,1))
ggsave(filename="unique_vs_new_genes.png", scale=1)
```

More info in: https://github.com/sanger-pathogens/Roary/blob/master/bin/create_pan_genome_plots.R


Now with the presence and absence of genes file
```
install.packages(c("pheatmap", "tidyverse"))

library(pheatmap)
library(tidyverse)

rtab <- read.delim("gene_presence_absence.Rtab", check.names = FALSE)

mat <- as.matrix(rtab)

pheatmap(mat,
         color = colorRampPalette(c("white", "darkgreen"))(100),
         cluster_rows = TRUE,
         cluster_cols = TRUE,
         show_rownames = FALSE,
         main = "Presencia/Ausencia de Genes")

```

### ML-Phylogeny (using iQtree)

If you want to run iQtree to select the best model and construct the phylogeny. This comand lines could help



You first need to build a multiple alignment

```
muscle -in SelectedProteins.fasta  -fastaout alnProts.fasta
```

Then run iQtree using multiple alignments

```
iqtree -s alnProts.fasta -m TEST -alrt 1000 -st AA > modeltest.txt


iqtree -s alnProts.fasta -m MODEL -alrt 1000 -nt AUTO -redo # if you want to use -alrt

iqtree -s alnProts.fasta -m MODEL -bb 1000 -redo #if you want to use bootstrap ultrafast

```



Tips:

| Opción    | ¿Busca modelo? | ¿Construye árbol? | Equivalente a               |
| --------- | -------------- | ----------------- | --------------------------- |
| `-m MF`   | ✅ Sí           | ❌ No              | Solo ModelFinder            |
| `-m TEST` | ✅ Sí           | ✅ Sí              | jModelTest/ProtTest         |
| `-m MFP`  | ✅ Sí           | ✅ Sí              | ModelFinder con particiones |




View the tree: https://itol.embl.de/




























